src/python/pants/option/config.py

```diff
@@ -230,10 +244,173 @@ def get_value(self, section: str, option: str) -> Optional[str]:
   def options(self, section: str) -> List[str]:
     return self.parser.options(section)
 
+  @property
   def defaults(self) -> Mapping[str, str]:
     return self.parser.defaults()
 
 
+_TomlPrimitve = Union[bool, int, float, str]
```

Comment:
We don't (yet?) account for TOML's native support of date times. We don't have any options that use that `type` so I left it off.